### PR TITLE
feat:add commands tree view

### PR DESCRIPTION
**Pull request by brianignacio5**
## Description

Add common commands to ESP-IDF Explorer tree view

Fixes VSC-1222

## Type of change

- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Click on "ESP-IDF Explorer" tab in the side panel with Espressif logo
2. Execute each of the common commands.
3. Observe results. Each command should behave as the same from Command palette.

- Expected behaviour:

- Expected output:

## How has this been tested?

Manual test by executing each action.

**Test Configuration**:
* ESP-IDF Version: 5.0
* OS (Windows,Linux and macOS): macOS

### CODE REVIEW

**Review by radurentea — APPROVED** [review]
LGTM

**Review by kolipakakondal — COMMENTED** [review]
Hi @brianignacio5 Thanks for the PR.

Quick things:

- Capitalize first letter in the word for command names (eg. Select Serial Port)
- Arrange command names based on how users will access them. For example, users would start with "Configure Extension," so we can keep this as the first item.
- Re-arrange the ESP-IDF Explorer layout based on how users will use it frequently. For instance, give more space to the "Commands" tree so that users can see all possible actions and minimize Documentation Search Results, Application Trace Archives, Rain Maker, etc.

**Comment by kolipakakondal** [comment]
Looks good now
Screenshot 2023-11-22 at 9 21 07 AM

File: src/cmdTreeView/cmdTreeDataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    if (element && element.commandType === CommandType.Category) {
+      return this.getAdvancedCommands();
+    }
+    return this.getInitialCommands();
+  }
+
+  private getAdvancedCommands() {
+    const createIdfTemplate = new CommandItem(
+      "Create ESP-IDF From Extension Templates",
+      CommandType.Command,
```
Comment (kolipakakondal) [inline src/cmdTreeView/cmdTreeDataProvider.ts:44]: Is this "Create Project Templates"?

File: src/cmdTreeView/cmdTreeDataProvider.ts [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+    const createVscodeFolder = new CommandItem(
+      "Add .vscode subdirectory files",
+      CommandType.Command,
+      "espIdf.createVsCodeFolder",
+      "add"
+    );
+    const devContainerFolder = new CommandItem(
+      "Add .devcontainer subdirectory files",
+      CommandType.Command,
```
Comment (kolipakakondal) [inline src/cmdTreeView/cmdTreeDataProvider.ts:101]: Add .devcontainer **S**ubdirectory **F**iles

File: src/cmdTreeView/cmdTreeDataProvider.ts [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+    const installMatterRequirements = new CommandItem(
+      "Install ESP-Matter Python Requirements",
+      CommandType.Command,
+      "espIdf.installEspMatterPyReqs",
+      "extensions"
+    );
+
+    const createVscodeFolder = new CommandItem(
+      "Add .vscode subdirectory files",
+      CommandType.Command,
```
Comment (kolipakakondal) [inline src/cmdTreeView/cmdTreeDataProvider.ts:95]: Add .vscode Subdirectory Files

File: src/cmdTreeView/cmdTreeDataProvider.ts [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+    );
+    const dfuFlash = new CommandItem(
+      "Flash with DFU",
+      CommandType.Command,
+      "espIdf.flashDFU",
+      "zap"
+    );
+    const websocketMonitor = new CommandItem(
+      "Launch Websocket server and IDF Monitor",
+      CommandType.Command,
```
Comment (kolipakakondal) [inline src/cmdTreeView/cmdTreeDataProvider.ts:131]: Launch Websocket **S**erver and IDF Monitor

**Comment by AndriiFilippov** [comment]
all other commands work fine 👍 
LGTM

**Review by AndriiFilippov — APPROVED** [review]

**Pull request merged**